include/linalg.hpp

```diff
@@ -740,7 +762,27 @@ namespace cytnx {
                                                  const double &err = 0., const bool &is_U = true,
                                                  const bool &is_vT = true,
                                                  const unsigned int &return_err = 0,
-                                                 const unsigned int &mindim = 1);
+                                                 const cytnx_uint64 &mindim = 1);
+
+    /**
+     * @brief Perform Singular-Value decomposition on a UniTensor with truncation and keep at most
+     * \p minblockdim singular values in each block.
+     * @details This function performs the Singular-Value decomposition of a UniTensor \p Tin and
+     * truncates the singular values. The result will depend on the rowrank of the
+     * UniTensor \p Tin. For each block, the minimum dimension can be chosen. This can be helpful to
+     * avoid loosing symmetry sectors in the truncated SVD. For more details, please refer to the
+     * documentation of the function Svd_truncate(const Tensor &Tin, const cytnx_uint64 &keepdim,
+     * const double &err, const bool &is_UvT, const unsigned int &return_err).
+     * @param[in] minblockdim a vector containing the minimum dimension of each block;
+     * alternatively, a vector with only one element can be given to have the same minblockdim for
+     * each block
+     * @see Svd_truncate(const Tensor &Tin, const cytnx_uint64 &keepdim, const double &err,
+     *                   const bool &is_UvT, const unsigned int &return_err)
+     */
+    std::vector<cytnx::UniTensor> Gesvd_truncate(
```

Comment:
What if provide `minblockdim` and `mindim` at the same time?

Reply:
`min_blockdim` and `mindim` can be given at the same time, first `min_blockdim` singular values are taken in each block, and if these are less then `mindim` then the difference is further added depending on the size of the singular values.